core/src/main/codegen/templates/Parser.jj

```diff
@@ -1661,7 +1663,12 @@ SqlNode SqlUpdate() :
         }
     ]
     [ [ <AS> ] alias = SimpleIdentifier() ]
-    <SET> id = SimpleIdentifier() {
+    (
+        <FROM> secondTable = TableRef()
+        [ [ <AS> ] secondAlias = SimpleIdentifier() ]
+    )?
```

Comment:
I ultimately wasn't sure how to do this using an injected method but I feel like this solution is fairly clean.